### PR TITLE
Issue #518: derive policy-aware business planning objectives

### PR DESCRIPTION
**Pull request by stranske**
> **Source:** Issue #518

## Summary
- add a reusable `BusinessPlanningObjectives` contract for policy-aware business planning posture
- derive deterministic business objectives from `BusinessTravelProfile` and optional `PolicyConstraintSet`
- add fixture-driven tests and a boundary doc for compliant-first, schedule-critical, and exception-aware cases

## Validation
- `python -m pytest -q tests/business/test_profile.py tests/business/test_policy_contracts.py tests/business/test_objective_derivation.py`
- `python -m pytest -q tests/contracts/test_objectives.py`
- `python -m mypy trip_planner/business tests/business/test_objective_derivation.py`

#### Scope
The business side should optimize toward policy fit and schedule protection before any final ranking layer is added. Once the repo has a business profile and policy-facing contracts, it still needs a deterministic intermediate layer that converts them into planning objectives, just as the leisure side needs `ItineraryObjectives` downstream from preference resolution.

Parent epic: #513

### Related Issues/PRs
- [#513](https://github.com/stranske/trip-planner/issues/513)
- [#505](https://github.com/stranske/trip-planner/issues/505)

#### Tasks
- [x] Define a `BusinessPlanningObjectives` contract capturing channel restrictions, schedule protection, comparable requirements, justification readiness, cost-control posture, comfort-floor protection, and exception-path posture.
- [x] Implement a derivation layer that maps `BusinessTravelProfile` and `PolicyConstraintSet` into `BusinessPlanningObjectives`.
- [x] Ensure the derivation layer distinguishes compliant-first objectives from policy-nearest fallback objectives when a clean compliant plan may not exist.
- [x] Preserve structured explanation output so later business-planning and policy-review UIs can explain why the planner favored certain options or channels.
- [x] Add tests covering at least: a straightforward compliant trip, a trip with tight schedule protection needs, and a trip that likely needs exception handling.
- [x] Document the boundary between business objective derivation and later business option ranking or search.

#### Acceptance criteria
- [x] The repo contains a reusable `BusinessPlanningObjectives` contract downstream from `BusinessTravelProfile` and policy constraints.
- [x] Different business fixtures produce materially different objective bundles in tests.
- [x] Objective derivation can represent both compliant-first and exception-aware planning postures.
- [x] Explanation output remains available for major objective categories.
- [x] Documentation clearly separates objective derivation from later ranking and live integration work.

### CODE REVIEW

File: trip_planner/business/objective_derivation.py [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+        "approval_roles:"
+        + ",".join(profile.approval_targets.approval_roles or ["none"]),
+        f"fallback_mode:{profile.exception_strategy.fallback_mode}",
+    ]
+    if constraint_set is not None:
+        explanations.append(f"policy_constraint_set:{constraint_set.policy_id}")
+        explanations.append(
+            "allowed_exception_types:"
+            + ",".join(constraint_set.allowed_exception_types or ["none"])
+        )
```
[the remark this answers is not in the document]
Comment (stranske) [reply trip_planner/business/objective_derivation.py]: Handled on the current branch head as well. The explanation-building path now sorts unordered inputs before joining them, and the merged required-channel list also comes from deterministic helpers.
